File: 정수론 및 조합론/2609/2609.c
#include <stdio.h>

int gcd(int, int);
int lcm(int, int);

int main()
{
   int a, b;
   scanf("%d %d", &a, &b);
   printf("%d\n%d", gcd(a, b), lcm(a, b));

   return 0;
}

int gcd(int a, int b)
{
   if (a < b) return gcd(b, a);
   if (a == b) return a;
   return gcd(a - b, b);
}
int lcm(int a, int b)
{
   int n = a;
   while (n % b)
      n += a;
   return n;
}

File: 정수론 및 조합론/2609/2609.py
def gcd(a, b):
   return gcd(b, a % b) if b else a
def lcm(a, b):
   return a * b // gcd(a, b)

a, b = map(int, input().split())
print(gcd(a, b))
print(lcm(a, b))